apps/video/test/marionette/lib/video_options_view.js

```diff
@@ -0,0 +1,99 @@
+/* global module */
+(function(module) {
+  'use strict';
+
+  /**
+   * @constructor
+   * @param {Marionette.Client} client Marionette client to use.
+   */
+  function VideoOptionsView(client) {
+    this.client = client.scope({
+    	searchTimeout: 20000
+  	});
+  }
+
+  module.exports = VideoOptionsView;
+
+  /**
+   * @const {Object}
+   */
+  VideoOptionsView.Selector = Object.freeze({
+    optionsView: '#options-view',
+    shareButton: '.single-share-button',
+    infoButton: '.single-info-button',
+    deleteButton: '.single-delete-button',
+    cancelButton: '.gaia-menu-cancel'
+  });
+
+  VideoOptionsView.prototype = {  
+    
+    client: null,
+
+    /**
+    * @return {Marionette.element} share button object
+    */
+  	get shareButton() {
+      return this.client.findElement(
+        VideoOptionsView.Selector.shareButton);
+    },
+
+    /**
+    *  @return {Marionette.element} more info button object
+    */
+    get infoButton() {
+      return this.client.findElement(
+        VideoOptionsView.Selector.infoButton);
+    },
+
+    /**
+    *  @return {Marionette.element} delete button object
+    */
+    get deleteButton() {
+      return this.client.findElement(
+        VideoOptionsView.Selector.deleteButton);
+    },
+
+    /**
+    *  @return {Marionette.element} cancel button object
+    */
+    get cancelButton() {
+      return this.client.findElement(
+        VideoOptionsView.Selector.cancelButton);
+    },
+
+    /**
+    * @return {function} video player option view class.
+    */
+    _createInfoView: function () {
+    var VideoInfoView = require('./video_info_view.js');
+    var newInfoView = new VideoInfoView(this.client);
+    return newInfoView;
+    },
+
+    /**
+    * @return {boolean} the display status of the options view
+    */
+    waitForOptionsViewDisplayed: function() {
+      return this.client.helper.waitForElement(
+      VideoOptionsView.Selector.optionsView).displayed();
+    },
+
+    tapShare: function() {
+      this.shareButton.click();
+    },
+
+    tapMoreInfo: function() {
+      this.infoButton.click();
+      this._createInfoView().waitForInfoViewDisplayed();
```

Comment:
I like how you have introduced 'waitForInfoViewDisplayed' in VideoInfoView. Using that in the test code before testing the video info view properties will prevent any timing issues. Since the test is calling 'waitForInfoViewDisplayed', 'tapMoreInfo' doesn't need to call it. And then '_createInfoView' is not needed :)